pkg/cmd/server/origin/controller/network_sdn.go

```diff
@@ -18,13 +21,20 @@ func (c *SDNControllerConfig) RunController(ctx ControllerContext) (bool, error)
 		return false, nil
 	}
 
+	networkClient := ctx.ClientBuilder.OpenshiftInternalNetworkClientOrDie(bootstrappolicy.InfraSDNControllerServiceAccountName)
+	networkInformers := networkinformers.NewSharedInformerFactory(networkClient, network.DefaultInformerResyncPeriod)
+
 	if err := sdnmaster.Start(
 		c.NetworkConfig,
-		ctx.ClientBuilder.OpenshiftInternalNetworkClientOrDie(bootstrappolicy.InfraSDNControllerServiceAccountName),
+		networkClient,
 		ctx.ClientBuilder.KubeInternalClientOrDie(bootstrappolicy.InfraSDNControllerServiceAccountName),
 		ctx.InternalKubeInformers,
+		networkInformers,
 	); err != nil {
 		return false, fmt.Errorf("failed to start SDN plugin controller: %v", err)
 	}
+
+	networkInformers.Start(utilwait.NeverStop)
```

Comment:
You have to wait for the informers to sync before people start acting on it, if you have any interactions between your controllers that can be destructive.

I.e. if you require that node A exist before hostsubnet B can exist, otherwise you delete hostsubnet B, if you don't wait for the informers to sync then your controller might fail to load nodes for a very long time, causing it to delete B (because no nodes have been loaded).

Unless you have a really good reason not to, your controllers need to wait for sync (like other controllers)